package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -673,10 +674,25 @@ def _read_frame(self, frame):
     def _copy_to_data(self):
         """assigns values to keys in data dictionary"""
 
-        if 'observables' in self._file:
-            for key in self._file['observables'].keys():
-                self.ts.data[key] = self._file['observables'][key][
-                    'value'][self._frame]
+        if "observables" in self._file:
+            for key in self._file["observables"].keys():
+                try:
+                    # if has value as subkey read directly into data
+                    if "value" in self._file["observables"][key]:
+                        self.ts.data[key] = self._file["observables"][key][
+                            "value"
+                        ][self._frame]
+                    # if value is not a subkey, read dict of subkeys
+                    else:
+                        for subkey in self._file["observables"][key].keys():
+                            self.ts.data[key + "/" + subkey] = self._file[
+                                "observables"
+                            ][key][subkey]["value"][self._frame]
+                except KeyError:
+                    warnings.warn(
```

Comment:
If it's not legal, shouldn't we be *failing*? Or is the reasoning that we don't overly care about observables as opposed to x/v/f?

Perhaps add a comment explaining your rationale.

Also, given that this is not tested, add a `#pragma: nocover` (or whatever you have to do to show that this code block isn't covered). It will tell anyone else reading the code that it's un-tested. 

Tested would be better, of course, but if we don't care about observables then I don't insist on a test.

Reply:
I'd be in favor of raising an Error here as well. Shall I create an additional `h5md` file which contains wrong data to test against this?

Reply:
That would be ideal!